src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs

```diff
@@ -130,17 +180,41 @@ public static Task<WwwAuthenticateParameters> CreateFromResourceResponseAsync(st
             return CreateFromResourceResponseAsync(resourceUri, default);
         }
 
+        /// <summary>
+        /// Create a list of the known authenticate parameters by attempting to call the resource unauthenticated, and analyzing the response.
+        /// </summary>
+        /// <param name="resourceUri">URI of the resource.</param>
+        /// <param name="cancellationToken">The cancellation token to cancel operation.</param>
+        /// <returns>a list of WWW-Authenticate Parameters extracted from a response to the unauthenticated call.</returns>
+        public static async Task<IEnumerable<WwwAuthenticateParameters>> CreateKnownParametersFromResourceResponseAsync(string resourceUri, CancellationToken cancellationToken = default)
```

Comment:
Grabs a parsed list of all of the known/supported authentication schemes. This is to help facilitate fallback scenarios when Pop is not supported.